bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/callmonitor/CallMonitor.java

```diff
@@ -0,0 +1,217 @@
+                    while (!interrupted) {
+                        try {
+                            String line = reader.readLine();
+                            if (line != null) {
+                                logger.debug("Received raw call string from fbox: {}", line);
+                                CallEvent ce = new CallEvent(line);
+                                handleCallEvent(ce);
+                            }
+                        } catch (IOException e) {
+                            if (interrupted) {
+                                logger.debug("Lost connection to Fritzbox because of an interrupt.");
+                            } else {
+                                handler.setStatusInfo(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                        "Lost connection to Fritz!Box: " + e.getMessage());
+                            }
+                            break;
+                        } finally {
+                            try {
+                                sleep(1000L);
+                            } catch (InterruptedException e) {
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        /**
+         * Close socket and stop running thread.
+         */
+        @Override
+        public void interrupt() {
+            interrupted = true;
+            if (socket != null) {
+                try {
+                    socket.close();
+                    logger.debug("Socket to FritzBox closed.");
+                } catch (IOException e) {
+                    logger.warn("Failed to close connection to FritzBox.", e);
+                }
+            } else {
+                logger.debug("Socket to FritzBox not open, therefore not closing it.");
+            }
+        }
+
+        /**
+         * Handle call event and update item as required.
+         *
+         * @param ce call event to process
+         */
+        private void handleCallEvent(CallEvent ce) {
+            if (ce.getCallType().equals("DISCONNECT")) {
+                // reset states of call monitor channels
+                handler.updateState(AVMFritzBindingConstants.CHANNEL_CALL_INCOMING, UnDefType.NULL);
+                handler.updateState(AVMFritzBindingConstants.CHANNEL_CALL_OUTGOING, UnDefType.NULL);
+                handler.updateState(AVMFritzBindingConstants.CHANNEL_CALL_ACTIVE, UnDefType.NULL);
+                handler.updateState(AVMFritzBindingConstants.CHANNEL_CALL_STATE,
+                        AVMFritzBindingConstants.CALL_STATE_IDLE);
+            } else if (ce.getCallType().equals("RING")) { // first event when call is incoming
+                StringListType state = new StringListType(ce.getInternalNo(), ce.getExternalNo());
+                handler.updateState(AVMFritzBindingConstants.CHANNEL_CALL_INCOMING, state);
```

Comment:
good point.

Reply:
Sry to be nit-picking on this. According our guidelines we should set the state to `UnDefType.UNDEF`.

Reply:
Which guideline is that? UNDEF is to be used in case the state cannot be mapped to any sensible value and NULL, if no state exists at all (yes, this might be debatable here, but it seems both values can be found arguments for). But if you prefer UNDEF, I can change it.

Reply:
Hm, ... I always thought the state should only be `NULL` after restart of OH and otherwise one always should use `UNDEF` if there is no valid data available to set.

Reply:
In most cases that is the case this way. But there are a few exceptional cases, where it is better to say that a channel simply does not have ANY state and then `NULL` is the better match. E.g. when a device operates in a mode, where the channel does not make any functional sense (yeah, ideally, the channel should then be dynamically completely removed).
But yeah, I am thus fine with `UNDEF` here for the call channels. In 1.x, the CallType had a dedicated `EMPTY` value, but this does not exist for us anymore.